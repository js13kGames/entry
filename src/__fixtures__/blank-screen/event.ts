import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 1,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/blank-screen",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "360b838";
