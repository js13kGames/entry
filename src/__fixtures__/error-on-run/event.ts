import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 6,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/error-on-run",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;
