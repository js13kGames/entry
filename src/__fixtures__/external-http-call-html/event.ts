import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 5,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/external-http-call-html",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "c9a43c6";
