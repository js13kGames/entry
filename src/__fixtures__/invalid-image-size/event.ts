import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 12,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/invalid-image-size",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "ce740ea";
