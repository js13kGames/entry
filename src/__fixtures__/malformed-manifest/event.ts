import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 10,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/malformed-manifest",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;
