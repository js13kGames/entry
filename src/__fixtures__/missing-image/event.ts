import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 11,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-image",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "400d963";
