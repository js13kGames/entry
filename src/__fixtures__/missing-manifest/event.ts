import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 9,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-manifest",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "de7f563";
