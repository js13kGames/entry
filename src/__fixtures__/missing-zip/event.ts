import { PullRequest } from "../../services/github";

export const installation = { id: 1319757 };

export const pullRequest: PullRequest = {
  number: 3,
  base: {
    repo: {
      name: "entry",
      owner: { login: "js13kgames" }
    }
  },
  head: {
    ref: "entry/missing-zip",
    repo: {
      name: "entry",
      owner: { login: "platane" }
    }
  }
} as any;

export const commitSha = "e60be98";
