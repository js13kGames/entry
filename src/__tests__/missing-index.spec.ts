import * as event from "../__fixtures__/missing-index/event";
import { bootstrap } from "./_bootstrap";

describe("integration  missing-index", () => {
  bootstrap(event);
});
