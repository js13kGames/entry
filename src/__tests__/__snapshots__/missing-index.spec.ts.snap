// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration  missing-index should analyze 1`] = `
Array [
  Object {
    "conclusion": "success",
    "name": "username-found",
    "repositoryName": "entry",
    "username": "platane",
  },
  Object {
    "conclusion": "success",
    "files": Array [
      "asset/image/1659821.jpeg",
      "asset/script.js",
      "asset/text.txt",
      "game.html",
      "image_large.jpg",
      "image_thumbnail.jpg",
      "manifest.json",
    ],
    "name": "manifest-found",
  },
  Object {
    "conclusion": "success",
    "manifest": Object {
      "description": "This entry should yield an error because the index.html is missing from the bundle",
      "name": "missing index",
    },
    "name": "manifest-read",
  },
  Object {
    "conclusion": "success",
    "gameName": "missing index",
    "name": "name-found",
  },
  Object {
    "conclusion": "success",
    "gameDescription": "This entry should yield an error because the index.html is missing from the bundle",
    "name": "description-found",
  },
  Object {
    "conclusion": "success",
    "images": Object {
      "image_large": "https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_200,w_300/81d8ad1eef5dbf6feda2665a0df3bf19.jpg",
      "image_thumbnail": "https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_100,w_100/61191dccc4c29b4d785cae9b2f2eae2d.jpg",
    },
    "name": "images-found",
  },
  Object {
    "branch": "entry/missing-index",
    "conclusion": "success",
    "name": "release-found",
    "releaseUrl": "https://github.com/Platane/entry/releases/tag/entry/missing-index-v1",
    "tagName": "entry/missing-index-v1",
  },
  Object {
    "assetFiles": Array [
      "bundle.zip",
    ],
    "bundleUrl": "http://js13kgames-entry-preview.s3-website-eu-west-1.amazonaws.com/3422b369f99ee2f4ac965ee30ef7bd0e/bundle.zip",
    "conclusion": "success",
    "name": "bundle-found",
  },
  Object {
    "conclusion": "success",
    "name": "bundle-unzipped",
  },
  Object {
    "bundleSize": 3857,
    "conclusion": "success",
    "name": "bundle-size",
    "sizeLimit": 13312,
  },
  Object {
    "bundleFiles": Array [
      "asset/text.txt",
      "asset/script.js",
      "asset/image/1659821.jpeg",
      "game.html",
    ],
    "conclusion": "failure",
    "deployUrl": undefined,
    "name": "index-found",
  },
]
`;

exports[`integration  missing-index should generate checks 1`] = `
Array [
  Object {
    "conclusion": "success",
    "name": "meta",
    "output": Object {
      "summary": "
## ‚úîÔ∏è manifest found
The pull requet contains a file named \`manifest.json\`

## ‚úîÔ∏è manifest parsed
The manifest was found :
\`\`\`json
{
  \\"name\\": \\"missing index\\",
  \\"description\\": \\"This entry should yield an error because the index.html is missing from the bundle\\"
}
\`\`\`

## ‚úîÔ∏è game info
The game name is \\"missing index\\"

The game description is:

This entry should yield an error because the index.html is missing from the bundle

## ‚úîÔ∏è game images
 - image_thumbnail

<img height=\\"100\\" src=\\"https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_100,w_100/61191dccc4c29b4d785cae9b2f2eae2d.jpg\\" />
 - image_large

<img height=\\"100\\" src=\\"https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_200,w_300/81d8ad1eef5dbf6feda2665a0df3bf19.jpg\\" />",
      "text": "",
      "title": "",
    },
  },
  Object {
    "conclusion": "failure",
    "name": "bundle",
    "output": Object {
      "summary": "
## ‚úîÔ∏è release
The release [entry/missing-index-v1](https://github.com/Platane/entry/releases/tag/entry/missing-index-v1) was found

## ‚úîÔ∏è bundle
The bundle was found

## ‚úîÔ∏è game info
The bundle was unzipped\\"

## ‚ùå index file
The bundle appear to not contain a \`index.html\` file
Those are the files found as assets:
\`\`\`
 - asset/text.txt
 - asset/script.js
 - asset/image/1659821.jpeg
 - game.html
\`\`\`

> Make sure the bundle contains an \`index.html\` file

## ‚úîÔ∏è bundle size
The bundle size is ok",
      "text": "",
      "title": "",
    },
  },
  Object {
    "conclusion": "failure",
    "name": "game",
    "output": Object {
      "summary": "
## ‚ùå deploy
The game could not be deployed, check the bundle report to see why.

> Make sure that the bundle is valid first.",
      "text": "",
      "title": "",
    },
  },
]
`;

exports[`integration  missing-index should generate report 1`] = `
"Hello üëã
I am a bot. My job is to review your submission.

I reviewed your latest release, [entry/missing-index-v1](https://github.com/Platane/entry/releases/tag/entry/missing-index-v1)
I could not found a index.html file in the zip archive.
I found:
\`\`\`
 - asset/text.txt
 - asset/script.js
 - asset/image/1659821.jpeg
 - game.html
\`\`\`

Can you make sure your game can be launched through a file named \`index.html\`?


<blockquote> <br> <img height=\\"80\\" width=\\"80\\" title=\\"thumbnail\\" src=\\"https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_100,w_100/61191dccc4c29b4d785cae9b2f2eae2d.jpg\\" /> <img height=\\"80\\" title=\\"large image\\" src=\\"https://res.cloudinary.com/dy3iszldp/image/upload/c_fit,h_200,w_300/81d8ad1eef5dbf6feda2665a0df3bf19.jpg\\" /> <div>  <a href=\\"undefined\\">   <strong>missing index</strong>  </a>  <i>   <span>&nbsp;&nbsp;by</span>   <a href=\\"https://github.com/platane\\">    platane   </a>  </i>  <br>  This entry should yield an error because the index.html is missing from the bundle </div> <br></blockquote>"
`;
